ddl/reorg.go

```diff
@@ -838,6 +835,55 @@ func (r *reorgHandler) RemoveDDLReorgHandle(job *model.Job, elements []*meta.Ele
 	return r.m.RemoveDDLReorgHandle(job, elements)
 }
 
+// CleanupDDLReorgHandles removes the job reorganization related handles.
+func CleanupDDLReorgHandles(job *model.Job, pool *sessionPool, concurrentDDL bool) {
+	if job != nil && !job.IsFinished() && !job.IsSynced() {
+		// Job is given, but it is neither finished nor synced; do nothing
+		return
+	}
+	se, err := pool.get()
+	if err != nil {
+		logutil.BgLogger().Info("CleanupDDLReorgHandles get sessionctx failed", zap.Error(err))
+		return
+	}
+	defer pool.put(se)
+
+	// Should there be any other options?
+	se.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
+	sess := newSession(se)
+	err = sess.begin()
+	if err != nil {
+		logutil.BgLogger().Info("CleanupDDLReorgHandles new session failed", zap.Error(err))
+		return
+	}
+	txn, err := sess.txn()
+	if err != nil {
+		logutil.BgLogger().Info("CleanupDDLReorgHandles new txn failed", zap.Error(err))
+		sess.rollback()
+	}
+	if concurrentDDL {
+		err = cleanDDLReorgHandles(sess, job)
```

Comment:
can the job be nil here?

Reply:
Hmm, during development it could, but with the three current entry points of CleaupDDLReorgHandles, job cannot be nil. But the code should be able to handle job == nil, just in case.